activation/activation_test.go

```diff
@@ -1104,21 +1113,36 @@ func TestBuilder_InitialProofGeneratedOnce(t *testing.T) {
 	assertLastAtx(require.New(t), tab.nodeID, types.BytesToHash(poetByte), atx, vPrevAtx, vPrevAtx, layersPerEpoch)
 
 	// GenerateProof() should not be called again
-	require.NoError(t, tab.generateInitialPost(context.Background()))
+	p, m, err = tab.generateInitialPost(context.Background())
+	require.NoError(t, err)
+	require.Nil(t, p)
+	require.Nil(t, m)
 }
 
 func TestBuilder_InitialPostIsPersisted(t *testing.T) {
 	tab := newTestBuilder(t, WithPoetConfig(PoetConfig{PhaseShift: layerDuration * 4}))
-	tab.mpost.EXPECT().GenerateProof(gomock.Any(), shared.ZeroChallenge, gomock.Any()).Return(&types.Post{}, &types.PostMetadata{}, nil)
-	require.NoError(t, tab.generateInitialPost(context.Background()))
+	tab.mpost.EXPECT().Config().AnyTimes().Return(PostConfig{})
+	tab.mpost.EXPECT().GenerateProof(gomock.Any(), shared.ZeroChallenge, gomock.Any()).Return(&types.Post{}, &types.PostMetadata{
+		Challenge: shared.ZeroChallenge,
+	}, nil)
+	p, m, err := tab.generateInitialPost(context.Background())
+	require.NoError(t, err)
+	require.NotNil(t, p)
+	require.NotNil(t, m)
```

Comment:
All these assertions aren't needed to verify that `generateInitialPost` behaves as expected in my opinion. The fat that `tab.mpost.GenerateProof()` wasn't called on the second call to `generateInitialPost` is already enough to proof that no additional proof was generated, i.e. `tab.initialPost` didn't change